### DISCUSSION TITLE
Consider throwing a `MethodNotAllowedHttpException` instead of `BadRequestHttpException` for `Controller::requirePostRequest`

### DISCUSSION THREAD

**Discussion by msbit** [craftcms/cms#13397] [Ideas] [answered]
For [`craft\web\Controller::requirePostRequest`](
https://github.com/craftcms/cms/blob/develop/src/web/Controller.php#L462-L467), as it's related specifically to the HTTP method expected, it seems to me a better match to throw a:

  `yii\web\MethodNotAllowedHttpException` (resulting in a HTTP 405 error)

instead of the more general:

  `yii\web\BadRequestHttpException`.

**Comment by brandonkelly** [comment, marked as answer]
Agree, that’s better. Just made that change for Craft 5 (77a88b31f6bf47e67ebc04604bde247d50708528).
